index.js

```diff
@@ -6,13 +6,153 @@ function ask(questionText) {
+let randNum = Math.floor((maxNum + minNum) / 2)
 
-start();
-
+//function that asks the user to guess a randomly generated number by the computer.
+async function reverseGame() {
+  //Initialize a different random number for this specific function. This number is generated randomly from 1-100.
+  let randNum = Math.floor(Math.random() * (maxNum - minNum + 1)) + minNum
+  //Take a guess from the user and parse that value as an integer because it comes in as a string.
+  let inputUser = await ask("I am thinking of a random number between 1 and 100 inclusive. Can you guess what it is? ")
+  inputUser = parseInt(inputUser)
+  //While loop that runs as long as the number guessed by the user is not equal to the actual number
+  while (inputUser !== randNum) {
+    //If the number guessed by user is greater than the actual number, print that the guess is too high and prompt the user to try again. Because this prompt takes user input again which comes in as a string, it must be parsed again. 
+    if (inputUser > randNum) {
+      console.log("Your number is too high!!")
+      inputUser = await ask("Try again: ")
+      inputUser = parseInt(inputUser)
+    }
+    //Same as the previous comment except vice versa. 
+    else if (inputUser < randNum) {
+      console.log("Your number is too low!!")
+      inputUser = await ask("Try again: ")
+      inputUser = parseInt(inputUser)
+    }
+  }
+  //If the number guessed by the user is equal to the actual mumber, print that and then exit the program.
+  console.log(`You guessed my number correctly! It's ${randNum}!`)
+  process.exit()
+}
+//Function that contains the code for a guessing game where the computer tries to guess a number the user inputs.
 async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
-  process.exit();
+    //Initialize a maximum range to be the number the user inputs. Parse the value and use that max to get a random integer using binary search.
+    maxNum = await ask("Choose a max range: ")
+    maxNum = parseInt(maxNum)
+    randNum = Math.floor((maxNum + minNum) / 2)
+    console.log("Choose a number between 1 and that number inclusive")
+    //Ask the user to choose a number within that range and then store it in a variable
+    let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
+    //Use the randomly generated number from line 47 and ask if that's the user's number.
+    let userInput = await ask(`Is your number ${randNum}? Type 'Y' or 'N': `)//<- this user input is refenced again in line 81.
+    
+    //Function that detects if the user is cheating and ends the program if he or she actually cheats.
+    async function rangeFunction() {
```

Comment:
async function rangeFunction() set within async function start() isn’t best practices. Ideally rangeFunction() is defined outside of start() and called within.